### ISSUE TITLE
Only allow a Classic Menu to be imported once (avoid duplication)

### ISSUE THREAD

**Issue by cbravobernal** [WordPress/gutenberg#43856]
## What problem does this address?

Right now, is quite easy to end up with duplications on Classic Menus. Each time you click on 'Import from Classic Menu', a new block navigation menu is created, with no indication of the origin of that menu.

![Screenshot 2022-09-05 at 13 56 40](https://user-images.githubusercontent.com/37012961/188443489-2e0ea5da-4359-404a-8427-5d2f2bc0fd60.jpg)

I did some experiments on #43580 with local storage, but, I think is not the best way to go, as the reference of menus imported would be done by user and browser, creating inconsistencies in any workflow.

## What is your proposed solution?

If #42809 is landed, I think I can use the slug as a reference for both classic menus and nav menus to determine which ones have been already imported, and then apply this design approach (kudos to @javierarce here)

![image](https://user-images.githubusercontent.com/37012961/188444398-55ab6266-e2b7-4d46-9b98-32d63375b4e7.png)

**Comment by talldan** [comment]
Another possibility is to make Classic Menu conversion much less prominent. Possibly even making it part of the Manage menus screen or a separate modal. They maybe have too much prominence, when in practice they're not something that can be used until they're converted into navigation menus.

**Comment by javierarce** [comment]
> Another possibility is to make Classic Menu conversion much less prominent. Possibly even making it part of the Manage menus screen or a separate modal. They maybe have too much prominence, when in practice they're not something that can be used until they're converted into navigation menus.

That would be really nice, the design above was the quickest solution to implement without having to add any extra interface, just some labels, and icons. 

If we went with a module-based solution, we could have one item that said "Import classic menus" (that obviously only appeared if you had one or more) and would simply show you a list of menus to import (ideally on batch).

Let me know if you want me to explore this solution.

**Comment by cbravobernal** [comment]
As we are close to the feature freeze, I guess this issue won't be ready for the next release. I don't have a strong opinion here so if you prefer to invest more time in UX considering better approaches feel free. 😄

**Comment by annezazu** [comment]
Noting that this came up in the [seventeenth call for testing for the FSE Outreach Program](https://make.wordpress.org/test/2022/09/15/fse-program-testing-call-17-guiding-the-gutenberg-gallery/#comment-2751) with one tester accidentally importing multiple times before realizing what went wrong.

**Comment by draganescu** [comment]
Is it really wrong if we mark in the DB the classic menus as imported? Then we can just go to Dan's idea above where they're accessible in a future manage menus screen.

**Comment by cbravobernal** [comment]
> Is it really wrong if we mark in the DB the classic menus as imported? Then we can just go to Dan's idea above where they're accessible in a future manage menus screen.

For me it is not, as it should not make the DB much bigger. I guess duplicating menus is already increasing more the DB than this solution.

**Comment by cbravobernal** [comment]
Navigation evolution has gone far from this issue.

**Comment by scruffian** [comment]
I think this is still relevant :)

**Comment by getdave** [comment]
I thought about this some more. Given that the relationship between Classic Menu and Navigation Menu Post is not maintained (i.e. it's a one time import) we shouldn't block the user from re-importing the same Classic Menu. After all they might 

- import Classic Menu
- make significant changes in the Navigation block
- want to import the Classic Menu again in a different block/Navigation Menu

Another option is simply [forcing each imported menu to have unique names](https://github.com/WordPress/gutenberg/issues/39200). That way if a user is importing repeatedly they'll get `Classic Menu`, `Classic Menu 1`, `Classic Menu 2`....etc.

**Comment by tyxla** [comment]
I agree with @getdave that there's no good reason to forbid this and his scenario is a valid one. I would rather prefer closing this as a "wontfix". @scruffian WDYT?

**Comment by youknowriad** [comment]
Closing per @tyxla's assessment.
